File: src/test/java/com/dotnomi/fabricdependencyinjection/java/testclasses/postconstruct/failing/FailingPostConstructService.java
package com.dotnomi.fabricdependencyinjection.java.testclasses.postconstruct.failing;

import com.dotnomi.fabricdependencyinjection.annotation.ModScoped;
import com.dotnomi.fabricdependencyinjection.annotation.PostConstruct;

@ModScoped
public class FailingPostConstructService {
  @PostConstruct
  private void initialize() {
    throw new RuntimeException("Initialization failed!");
  }
}

File: src/test/java/com/dotnomi/fabricdependencyinjection/kotlin/testclasses/postconstruct/failing/FailingPostConstructService.kt
package com.dotnomi.fabricdependencyinjection.kotlin.testclasses.postconstruct.failing

import com.dotnomi.fabricdependencyinjection.annotation.ModScoped
import com.dotnomi.fabricdependencyinjection.annotation.PostConstruct

@ModScoped
class FailingPostConstructService {
    @PostConstruct
    private fun initialize() {
        throw RuntimeException("Initialization failed!")
    }
}
